src/Stratis.Bitcoin.Features.Miner/Staking/PosMinting.cs

```diff
@@ -760,6 +745,49 @@ private void CheckStake(Block block, ChainedHeader chainTip)
             return true;
         }
 
+        internal Transaction PrepareCoinStakeTransactions(
+            int currentChainHeight,
+            CoinstakeContext coinstakeContext,
+            long coinstakeInputValue,
+            int utxosCount,
+            long amountStaked)
+        {
+            // Split stake into SplitFactor utxos if above threshold.
+            const int SplitFactor = 8;
```

Comment:
Why 8 is this a random number?

Reply:
Discussed with @Aprogiena, 8 is the biggest number that allows you to split a coin without exceeding the max size for the coinstake transaction

Reply:
Also this constant to the class so that you can use it in the tests